slsim/Sources/source.py

```diff
@@ -452,20 +458,28 @@ def extended_source_position(self, center_lens, draw_area):
         :param center_lens: center of the deflector.
          Eg: np.array([center_x_lens, center_y_lens])
         :param draw_area: The area of the test region from which we randomly draw a
-         source position. Eg: 4*pi.
+        source position. Eg: 4*pi.
         :return: [x_pos, y_pos]
         """
 
         if not hasattr(self, "_center_source"):
-            # Define the radius of the test area circle
-            test_area_radius = np.sqrt(draw_area / np.pi)
-            # Randomly generate a radius within the test area circle
-            r = np.sqrt(np.random.random()) * test_area_radius
-            theta = 2 * np.pi * np.random.random()
-            # Convert polar coordinates to cartesian coordinates
-            center_x_source = center_lens[0] + r * np.cos(theta)
-            center_y_source = center_lens[1] + r * np.sin(theta)
-            self._center_source = np.array([center_x_source, center_y_source])
+            # 1) Check if the user has provided center_x/center_y in source_dict
+            if (
+                "center_x" in self.source_dict.colnames
```

Comment:
I am not much a fan of this specific line for a specific source class. Can we make the center_x center_y  to become part of self._center_source?